zebrad/src/components/sync/downloads.rs

```diff
@@ -175,7 +175,7 @@ where
                         .next()
                         .expect("successful response has the block in it")
                 } else {
-                    unreachable!("wrong response to block request");
+                    return Err("wrong response to block request".into());
```

Comment:
Current `request_genesis_is_rate_limited()` test is getting into this unreachable so i had to change it to an Error.